src/api/namespaces/utils.ts

```diff
@@ -11,14 +12,16 @@ export type FilterDefinition = {
     }
 };
 
-export function mergeRequestData(req: express.Request): {[key: string]: any} {
+export type RequestParams = {[key: string]: any};
+
+export function mergeRequestData(req: express.Request): RequestParams {
     return Object.assign({}, req.query || {}, req.body || {});
 }
 
-export function filterQueryArgs(req: express.Request, filter: FilterDefinition, keyType: string = null): {[key: string]: any} {
+export function filterQueryArgs(req: RequestParams | express.Request, filter: FilterDefinition, keyType: string = null): RequestParams {
     const keys = Object.keys(filter);
-    const result: {[key: string]: any} = {};
-    const merged = mergeRequestData(req);
+    const result: RequestParams = {};
+    const merged = req.socket instanceof Socket ? mergeRequestData(req as express.Request) : req;
```

Comment:
Why do we need this condition here?

Reply:
This only needs to be here until all routes use the new handler. At the moment req can either be the express.request or in the new version an object of all paramaters